### PR TITLE
make title, message optional.

### PR DESCRIPTION
**Pull request by MoElnaggar14**
add UIStackView that contain titleLabel, messageLabel to handle nil or empty title, message

### CODE REVIEW

**Comment by MoElnaggar14** [comment]
@dkcas11 please, can you check this PR?

File: CRNotifications/Classes/CRNotification.swift [hunk 1]
```diff
@@ -172,7 +171,11 @@ public class CRNotificationView: UIView, CRNotification {
     }
     
     @objc internal func dismissNotificationOnTap() {
-        delegate?.onNotificationTap(type: type,title: titleLabel.text, message: messageLabel.text)
+        guard let onClickDelegate = onClickDelegate else {
```
Comment (dkcas11) [inline CRNotifications/Classes/CRNotification.swift:174]: Redundant.

File: CRNotifications/Classes/CRNotification.swift [hunk 2]
```diff
@@ -34,38 +34,42 @@ public class CRNotificationView: UIView, CRNotification {
         label.translatesAutoresizingMaskIntoConstraints = false
         label.font = UIFont.systemFont(ofSize: 13, weight: .semibold)
         label.textColor = .white
-        label.numberOfLines = 2
+        label.numberOfLines = 0
```
Comment (dkcas11) [inline CRNotifications/Classes/CRNotification.swift:37]: The purpose of notifications is to inform of lightweight information. If your text cannot fit within 2 lines you are doing something wrong and should use another approach to communicate your information to your user, like an alert.

File: CRNotifications/Classes/CRNotifications.swift [hunk 3]
```diff
@@ -9,7 +9,7 @@
 import UIKit
 
 public protocol CRNotificationDelegate {
-    func onNotificationTap(type: CRNotificationType, title: String?, message: String?)
+    func onNotificationTap()
```
Comment (dkcas11) [inline CRNotifications/Classes/CRNotifications.swift:12]: Whats the reason for removing this entirely?

File: CRNotifications/Classes/CRNotification.swift [hunk 4]
```diff
@@ -34,38 +34,42 @@ public class CRNotificationView: UIView, CRNotification {
...
+        stackView.distribution = .fillProportionally
+        stackView.spacing = 0.0
+        return stackView
+    }()
+    
 	
     private var completion: () -> () = {}
-    private var type: CRNotificationType
-    public var delegate: CRNotificationDelegate?
+    public var onClickDelegate: CRNotificationDelegate?
```
Comment (dkcas11) [inline CRNotifications/Classes/CRNotification.swift:53]: Why is the delegate property renamed? There is no point in limiting the delegate property to such a narrow purpose. When you need other delegate methods you will need new specific protocols. It just polutes the codebase.

**Review by dkcas11 — CHANGES_REQUESTED** [review]

**Comment by dkcas11** [comment]
There is simply too big of a change in the API to consider this PR at its current state. Most of the stuff has recently been added by request of other users so this will be a breaking change in many ways.
It seems like this serves a very specific use case rather than appealing to a broader audience.
In regards to the amount of displayed text this has already been discussed a few times, such as here: https://github.com/dkcas11/CRNotifications/issues/6
Im sorry, but i think its best to fork the repo and adjust the implementation for your needs and embed the framework from there :-)

**Pull request closed without merging**